src/blockdb/wrapper.cpp

```diff
@@ -201,6 +203,27 @@ void SyncStorage(const CChainParams &chainparams)
                 }
             }
             setDirtyBlockIndex.insert(index);
+            blocksToRemove.push_back(index);
+            if(blocksToRemove.size() % 10000 == 0)
+            {
+                CDBBatch batch(*pblockdb);
+                for(auto removeIndex : blocksToRemove)
+                {
+                  std::ostringstream key;
+                  key << removeIndex->GetBlockTime() << ":" << removeIndex->GetBlockHash().ToString();
+                  batch.Erase(key.str());
+                }
+                pblockdb->WriteBatch(batch);
+                // you must use NULL here, not nullptr
+                CBlockIndex* indexfront = blocksToRemove.front();
+                std::ostringstream frontkey;
```

Comment:
when using a pointer we prefix with a "p"   , so pindexfront .  vector is v, so vBlocksToRemove, set , is set, map is map etc., number is n.